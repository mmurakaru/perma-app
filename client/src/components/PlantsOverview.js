import styled from 'styled-components/macro'
import PropTypes from 'prop-types'
import logo from '../assets/perma_logo.svg'

PlantsOverview.propTypes = {
    plants: PropTypes.array,
}

export default function PlantsOverview({ plants, handleClick }) {
    return (
        <ThumbnailWrapper>
            {plants.length > 0 ? (
                plants?.map(({ id, common_name, image_url }) => (
                    <PlantThumbnail onClick={() => handleClick(id)} key={id}>
                        <img src={image_url} alt="" />
                        <h2>{common_name}</h2>
                    </PlantThumbnail>
                ))
            ) : (
                <ErrorMsgStyled>
                    Whoops! We couldn't find the plant you were looking for.
                </ErrorMsgStyled>
            )}
        </ThumbnailWrapper>
    )
}

const PlantThumbnail = styled.section`
    img {
        height: 100px;
        width: 156px;
        object-fit: cover;
        border-radius: 5px;
        background: var(--darkGrey);
        background-image: url(${logo});
        background-repeat: no-repeat;
    }
`

const ThumbnailWrapper = styled.div`
    margin-top: 20px;
    height: 70vh;
    display: grid;
    grid-template-columns: 1fr 1fr;
    grid-auto-rows: min-content;
    place-items: center;
    gap: 20px;
    overflow: auto;
`

const ErrorMsgStyled = styled.h2`
    position: absolute;
    color: var(--mediumGreen);
`
