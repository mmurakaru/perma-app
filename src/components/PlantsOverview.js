import styled from 'styled-components/macro'
import PropTypes from 'prop-types'
import { Link } from 'react-router-dom'

PlantsOverview.propTypes = {
    plants: PropTypes.array,
}

export default function PlantsOverview({ plants, handleClick }) {
    return (
        <ThumbnailWrapper>
            {plants.length > 0 ? (
                plants?.map(({ id, common_name, image_url }) => (
                    <Link to="/plant" className="link">
                        <PlantThumbnail
                            onClick={() => handleClick(id)}
                            key={id}
                        >
                            <img src={image_url} alt="" />
                            <h2>{common_name}</h2>
                        </PlantThumbnail>
                    </Link>
                ))
            ) : (
                <ErrormsgStyled>
                    Whoops! We couldn't find the plant you were looking for.
                </ErrormsgStyled>
            )}
        </ThumbnailWrapper>
    )
}

const PlantThumbnail = styled.section`
    img {
        height: 100px;
        width: 156px;
        object-fit: cover;
        border-radius: 5px;
    }
`

const ThumbnailWrapper = styled.div`
    margin-top: 20px;
    height: 600px;
    display: grid;
    grid-template-columns: 1fr 1fr;
    grid-auto-rows: min-content;
    place-items: center;
    gap: 20px;
    overflow: auto;

    .link {
        color: #4a4a4a;
        text-decoration: none;
    }
`

const ErrormsgStyled = styled.h2`
    position: absolute;
    color: #aabb97;
`
