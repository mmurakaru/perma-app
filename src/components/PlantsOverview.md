PlantsOverview example:

```js
<PlantsOverview
    plants={[
        {
            id: 1,
            common_name: 'Coconut palm',
            image_url: 'https://source.unsplash.com/random/life',
        },
        {
            id: 2,
            common_name: 'Strawberry',
            image_url: 'https://source.unsplash.com/random/plant',
        },
        {
            id: 3,
            common_name: 'Raspberry',
            image_url: 'https://source.unsplash.com/random/barcelona',
        },
        {
            id: 4,
            common_name: 'Elephant Gras',
            image_url: 'https://source.unsplash.com/random/forest',
        },
        {
            id: 5,
            common_name: 'Evengreen oak',
            image_url: 'https://source.unsplash.com/random/jungle',
        },
        {
            id: 6,
            common_name: 'Lilie',
            image_url: 'https://source.unsplash.com/random/green',
        },
        {
            id: 7,
            common_name: 'Palm',
            image_url: 'https://source.unsplash.com/random/office',
        },
        {
            id: 8,
            common_name: 'Basel',
            image_url: 'https://source.unsplash.com/random/earth',
        },
        {
            id: 9,
            common_name: 'Coconut palm',
            image_url: 'https://source.unsplash.com/random/life',
        },
        {
            id: 10,
            common_name: 'Strawberry',
            image_url: 'https://source.unsplash.com/random/plant',
        },
        {
            id: 11,
            common_name: 'Raspberry',
            image_url: 'https://source.unsplash.com/random/barcelona',
        },
        {
            id: 12,
            common_name: 'Elephant Gras',
            image_url: 'https://source.unsplash.com/random/forest',
        },
        {
            id: 13,
            common_name: 'Evengreen oak',
            image_url: 'https://source.unsplash.com/random/jungle',
        },
        {
            id: 14,
            common_name: 'Lilie',
            image_url: 'https://source.unsplash.com/random/green',
        },
        {
            id: 15,
            common_name: 'Palm',
            image_url: 'https://source.unsplash.com/random/office',
        },
        {
            id: 16,
            common_name: 'Basel',
            image_url: 'https://source.unsplash.com/random/earth',
        },
    ]}
/>
```
