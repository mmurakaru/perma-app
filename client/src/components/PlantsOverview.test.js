import PlantsOverview from './PlantsOverview'
import { render } from '@testing-library/react'

describe('Overview', () => {
    it('renders correctly', () => {
        const { container } = render(
            <PlantsOverview
                plants={[
                    {
                        id: 1,
                        common_name: 'Coconut palm',
                        image_url:
                            'https://bs.floristic.org/image/o/63073d2fbf45b90701279405ecc2eec0272906ed',
                    },
                    {
                        id: 2,
                        common_name: 'Strawberry',
                        image_url:
                            'https://bs.floristic.org/image/o/63073d2fbf45b90701279405ecc2eec0272906ed',
                    },
                ]}
            />
        )
        expect(container.firstChild).toMatchSnapshot()
    })
})
