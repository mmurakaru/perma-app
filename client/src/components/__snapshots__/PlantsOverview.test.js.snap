// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Overview renders correctly 1`] = `
.c1 img {
  height: 100px;
  width: 156px;
  object-fit: cover;
  border-radius: 5px;
  background: var(--darkGrey);
  background-image: url(perma_logo.svg);
  background-repeat: no-repeat;
}

.c0 {
  margin-top: 20px;
  height: 70vh;
  display: grid;
  grid-template-columns: 1fr 1fr;
  grid-auto-rows: min-content;
  place-items: center;
  gap: 20px;
  overflow: auto;
}

<div
  class="c0"
>
  <section
    class="c1"
  >
    <img
      alt=""
      src="https://bs.floristic.org/image/o/63073d2fbf45b90701279405ecc2eec0272906ed"
    />
    <h2>
      Coconut palm
    </h2>
  </section>
  <section
    class="c1"
  >
    <img
      alt=""
      src="https://bs.floristic.org/image/o/63073d2fbf45b90701279405ecc2eec0272906ed"
    />
    <h2>
      Strawberry
    </h2>
  </section>
</div>
`;
